pallets/route-executor/src/lib.rs

```diff
@@ -465,3 +702,26 @@ macro_rules! handle_execution_error {
 		}
 	}};
 }
+
+impl<T: Config> RouteProvider<T::AssetId> for Pallet<T> {
+	fn get(asset_pair: AssetPair<T::AssetId>) -> Vec<Trade<T::AssetId>> {
+		let onchain_route = Routes::<T>::get(asset_pair.ordered_pair());
+
+		let default_route = vec![Trade {
```

Comment:
nice, so you are keeping our current fallback alive

Reply:
that probably means that you should actually test against it if the omnipool route is not already set in router and not just take any route